query/undefine.rs

```diff
@@ -4,22 +4,1137 @@
+        ),
+        TypeDeleteError(
+            7,
+            "Error removing type '{label}'.\nSource:\n{declaration}",
+            label: Label<'static>,
+            declaration: TypeQLLabel,
+            ( typedb_source: ConceptWriteError )
+        ),
+        ValueTypeSymbolResolution(
+            8,
+            "Error resolving value type in undefine query.",
+            ( typedb_source: SymbolResolutionError )
+        ),
+        UnsetSupertypeError(
+            9,
+            "Undefining supertype failed.\nSource: {sub}",
+            sub: TypeQLSub,
+            ( typedb_source: ConceptWriteError )
+        ),
+        UnsetOwnsError(
+            10,
+            "Undefining owns failed.\nSource: {owns}",
+            owns: TypeQLOwns,
+            ( typedb_source: ConceptWriteError )
+        ),
+        DeleteRoleTypeError(
+            11,
+            "Undefining relates (role type) failed.\nSource: {relates}",
+            relates: TypeQLRelates,
+            ( typedb_source: ConceptWriteError )
+        ),
+        UnsetPlaysError(
+            12,
+            "Undefining plays failed.\nSource: {plays}",
+            plays: TypeQLPlays,
+            ( typedb_source: ConceptWriteError )
+        ),
+        UnsetValueTypeError(
+            13,
+            "Undefining '{label}' to have value type '{value_type}' failed.",
+            label: Label<'static>,
+            value_type: ValueType,
+            ( typedb_source: ConceptWriteError )
+        ),
+        UnsetTypeAnnotationError(
+            14,
+            "Undefining '{label}' to have annotation '{annotation_category}' failed.\nSource:\n{declaration}",
+            label: Label<'static>,
+            annotation_category: AnnotationCategory,
+            declaration: AnnotationType,
+            ( typedb_source: ConceptWriteError )
+        ),
+        UnsetCapabilityAnnotationError(
+            15,
+            "Undefining '{annotation_category}' failed.\nSource:\n{declaration}",
+            annotation_category: AnnotationCategory,
+            declaration: AnnotationCapability,
+            ( typedb_source: ConceptWriteError )
+        ),
+        UnsetRelatesSpecialiseError(
+            16,
+            "For relation type '{type_}', undefining `relates as' failed.\nSource:\n{declaration}",
+            type_: Label<'static>,
+            declaration: Specialise,
+            ( typedb_source: ConceptWriteError )
+        ),
+        TypeCannotHaveCapability(
+            17,
+            "Invalid undefine - the type '{type_}' of kind '{kind}', which cannot have: '{capability}'. Maybe you wanted to undefine something else?",
+            type_: Label<'static>,
+            kind: Kind,
+            capability: CapabilityType
+        ),
+        TypeSubNotDefined(
+            18,
+            "Undefining 'sub {supertype}' for type '{type_}' failed since there is no defined '{type_} sub {supertype}'.\nSource:\n{declaration}",
```

Comment:
We should generally not have constant strings that we know exist elsewhere - for `sub` should be from a TypeQL Token, same with `as`, `relates`... etc.

Reply:
It kinda makes sense (ignoring the fact that you were the one who introduced these strings first :wink:), although it makes the error message reading even messier, and we won't probably put whole error messages in our bdd checks. I can refactor it as I agree it's cleaner, but it doesn't feel that good.

Reply:
I have pushed an example of what I can do for `sub`, please check it out!

Reply:
Yeah i know i was rushing, I think there probably is a better strategy overall where we tie the problem back to an original  TypeQL query part. Although that isn't really always possible!